annotations/views/project_views.py

```diff
@@ -60,6 +60,30 @@ def add_text(self, request, pk=None):
         serializer = ProjectSerializer(project)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['POST'], url_name='removetext')
+    def remove_text(self, request, pk=None):
+        text_id = request.data['text_id']
+        repo_id = request.data['repository_id']
+
+        repository = get_object_or_404(Repository, pk=repo_id)
+        project = get_object_or_404(TextCollection, pk=pk)
+        manager = repository.manager(request.user)
+        resource = manager.resource(id=int(text_id))
+
+        defaults = {
```

Comment:
None of defaults are required. What defaults would be needed to delete something? You only need an id. See comment below about copying code.